### PR TITLE
Feat/watch challenges

### PR DESCRIPTION
**Pull request by Bouncey**
Good news Everyone!

This PR adds development functionality not seen in these parts for maybe two years...

Auto-challenge-update!

Once you edit a challenge markdown file, the new file is passed to gatsby to update the client you have running in your browser!

No need to refresh or stop/start the development environment, it is all taken care of for you!

:tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: 
:sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: :sparkles: 

---

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.

### CODE REVIEW

File: client/utils/buildChallenges.js [hunk 1]
```diff
@@ -10,8 +14,16 @@ const nameify = utils.nameify;
 const arrToString = arr =>
   Array.isArray(arr) ? arr.join('\n') : _.toString(arr);
 
+exports.replaceChallengeNode = function replaceChallengeNode(fullFilePath) {
+  const relativeChallengePath = fullFilePath.replace(
```
Comment (ValeraS) [inline client/utils/buildChallenges.js]: Is `fullFilePath` OS-specific?

File: package.json [hunk 2]
```diff
@@ -4,14 +4,14 @@
   "scripts": {
     "postinstall": "npm run bootstrap",
     "prebootstrap": "npm run ensure-env",
-    "bootstrap": "lerna bootstrap && lerna run build --scope @freecodecamp/curriculum",
```
Comment (ValeraS) [inline package.json]: Does`seed: challenges` use builded curriculum?

File: client/plugins/fcc-source-challenges/gatsby-node.js [hunk 3]
```diff
@@ -1,31 +1,67 @@
...
-"source" is a required option for fcc-source-challenges. It must be a function
-that delivers challenge files to the plugin
-      `);
+    "onSourceChange" is a required option for fcc-source-challenges. It must be
+    a function that delivers a new challenge object to the plugin
+    `);
   }
-  // TODO: Add live seed updates
   const { createNode } = actions;
+  const watcher = chokidar.watch(curriculumPath, {
```
[the remark this answers is not in the document]
Comment (ValeraS) [reply client/plugins/fcc-source-challenges/gatsby-node.js]: Can we remove `ignored` now?

File: package.json (hunk 2 above)
Comment (ValeraS) [reply package.json]: Curriculum package should be deleted from   seed `package.json`

File: client/plugins/fcc-source-challenges/gatsby-node.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bouncey) [reply client/plugins/fcc-source-challenges/gatsby-node.js]: We still need to ignore dot files, I have removed the `node_modules` check though.

File: package.json (hunk 2 above)
Comment (Bouncey) [reply package.json]: Removed from `seed` and `api-server`

**Pull request merged**